server/engine/src/main/java/org/eclipse/lsp/cobol/service/AnalysisService.java

```diff
@@ -225,7 +226,14 @@ public Node retrieveAnalysis(String uri, String text) {
     if (!documentModel.isDocumentSynced()) {
       analyzeDocumentWithCopybooks(uri, text);
     }
-    return documentService.get(uri).getAnalysisResult().getRootNode();
+    return Optional.ofNullable(documentService.get(uri))
+        .map(CobolDocumentModel::getAnalysisResult)
+        .map(AnalysisResult::getRootNode).orElseGet(() -> {
+          if (isCopybook(uri, text)) {
+            communications.notifyGeneralMessage(MessageType.Info, "Cannot retrieve outline tree because file was treated as a copybook");
```

Comment:
@zeibura please check this message
It will appear as a VSCode info box when a user tries to generate a graph from CCF for the copybook. Sometimes some COBOL programs are treated as a copybook and the user has the ability to try to build a graph and this message is to inform the user that it is not possible.